deepchecks/utils/ipython.py

```diff
@@ -304,121 +267,511 @@ def __exit__(self, *args, **kwargs):
+    'anyway. Therefore, if you encounter any problems with '
+    'interactive check/suite output, try to use instead '
+    'non-interactive check/suite results output, it can be '
+    'done in the next way:\n\n'
+    '>>> result = Check().run(...) # or Suite().run(...)\n'
+    '>>> result.show(as_widget=False)'
+)
+
+
+_EXTENSION_WARNING = (
+    '"{extension}" extension is not installed/enabled, '
+    'interactive check/suite output will not work for you{additional}. '
+    'In order to enable extension run next command "{enable_cmd}". '
+    'In order to install extension follow next instructions:\n{install_instructions}'
+)
+
+
+@lru_cache(maxsize=None)
+def _jupyterlab_widgets_warning(**kwargs):
+    warnings.filterwarnings(
+        action='once',
+        message=r'"@jupyter-widgets/jupyterlab-manager" extension is not installed/enabled.*',
+    )
+    warning_kwargs = dict(
+        extension='@jupyter-widgets/jupyterlab-manager',
+        enable_cmd='jupyter labextension enable @jupyter-widgets/jupyterlab-manager',
+        # TODO:
+        # I did not find any good instruction on internet
+        # probably better to write our own instruction for this
+        install_instructions=(
+            '+ https://pypi.org/project/jupyterlab-widgets/\n'
+            '+ https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#installing-extensions\n'
+        )
+    )
+    warning_kwargs.update(**kwargs)
+    return _EXTENSION_WARNING.format(**warning_kwargs)
+
+
+@lru_cache(maxsize=None)
+def _jupyterlab_plotly_warning(**kwargs):
+    warnings.filterwarnings(
+        action='once',
+        message=r'"jupyterlab-plotly" extension is not installed/enabled.*',
+    )
+    warning_kwargs = dict(
+        extension='jupyterlab-plotly',
+        enable_cmd='jupyter labextension enable jupyterlab-plotly',
+        # TODO:
+        # I did not find any good instruction on internet
+        # probably better to write our own instruction for this
+        install_instructions=(
+            '+ https://jupyterlab.readthedocs.io/en/stable/user/extensions.html#installing-extensions\n'
+        )
+    )
+    warning_kwargs.update(**kwargs)
+    return _EXTENSION_WARNING.format(**warning_kwargs)
+
+
+@lru_cache(maxsize=None)
+def _nbclassic_widgets_warning(**kwargs):
+    warnings.filterwarnings(
+        action='once',
+        message=r'"jupyter-js-widgets" extension is not installed/enabled.*',
+    )
+    warning_kwargs = dict(
+        extension='jupyter-js-widgets',
+        enable_cmd='jupyter nbextension enable jupyter-js-widgets/extension',
+        # TODO:
+        # I did not find any good instruction on internet
+        # probably better to write our own instruction for this
+        install_instructions='+ https://pypi.org/project/jupyter-js-widgets-nbextension/\n'
+    )
+    warning_kwargs.update(**kwargs)
+    return _EXTENSION_WARNING.format(**warning_kwargs)
+
+
+def is_widgets_enabled() -> bool:
```

Comment:
seems like a lot of duplicated code

Reply:
is_widgets_enabled and is_interactive_output_use_possible

Reply:
they are doing different things
1. verifies whether the use of interactive outputs is possible (whether nbclassic/jupyterlab widgets and jupyterlab plotly extensions are enabled)
3. verifies whether the use of ipywidgets is possible (whether nbclassic/jupyterlab widgets extension are enabled)

Reply:
what's the difference between interactive output and ipywidgets? is there another kind of interactive output?

Reply:
still many of this code. is duplicated

Reply:
> what's the difference between interactive output and ipywidgets? is there another kind of interactive output?

under interactive output I mean, situation when we use plotly and ipywidgets to display check/suite results

Reply:
>still many of this code is duplicated

sorry I do not get you
could you be more specific, please?

Reply:
so `is_interactive_output_use_possible` checks plotly + widgets and `is_widgets_enabled` checks only widgets?

Reply:
> so is_interactive_output_use_possible checks plotly + widgets and is_widgets_enabled checks only widgets?

yes,
we use `ipywidgets` for progress bars and they definitely do not require `jupyterlab-plotly` extension to work